src/network/networkd-mptcp.c

```diff
@@ -0,0 +1,388 @@
+                link->mp_tcp_configured = true;
+        else
+                log_link_debug(link, "Configuring MPTCP");
+
+        return 0;
+}
+
+int mp_tcp_section_verify(MPTCP *mp_tcp) {
+        assert(mp_tcp);
+
+        if (section_is_invalid(mp_tcp->section))
+                return -EINVAL;
+
+        if (mp_tcp->family == AF_UNSPEC)
+                return log_warning_errno(SYNTHETIC_ERRNO(EINVAL),
+                                         "%s: [MPTCP] section without Address= field configured. "
+                                         "Ignoring [MPTCP] section from line %u.",
+                                         mp_tcp->section->filename, mp_tcp->section->line);
+
+        return 0;
+}
+
+int config_parse_mp_tcp_id(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(mp_tcp_free_or_set_invalidp) MPTCP *mp_tcp = NULL;
+        Network *network = data;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = mp_tcp_new_static(network, filename, section_line, &mp_tcp);
+        if (r < 0)
+                return log_oom();
+
+        if (isempty(rvalue)) {
+                mp_tcp->id_is_set = false;
+
+                TAKE_PTR(mp_tcp);
+                return 0;
+        }
+
+        r = safe_atou8(rvalue, &mp_tcp->id);
+        if (r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Failed to parse MPTCP '%s=', ignoring assignment: %s", lvalue, rvalue);
+                return 0;
+        }
+
+        mp_tcp->id_is_set = true;
+
+        TAKE_PTR(mp_tcp);
+        return 0;
+}
+
+int config_parse_mp_tcp_address(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(mp_tcp_free_or_set_invalidp) MPTCP *mp_tcp = NULL;
+        Network *network = data;
+        int r, f;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = mp_tcp_new_static(network, filename, section_line, &mp_tcp);
+        if (r < 0)
+                return log_oom();
+
+        if (isempty(rvalue)) {
+                mp_tcp->address = IN_ADDR_NULL;
+                mp_tcp->family = AF_UNSPEC;
+                TAKE_PTR(mp_tcp);
+                return 0;
+        }
+
+        r = in_addr_from_string_auto(rvalue, &f, &mp_tcp->address);
```

Comment:
I think it is better to change the type `mp_tcp->family` to `int`, and simplify this line as follows.
```
r = in_addr_from_string_auto(rvalue, &mp_tcp->family, &mp_tcp->address);
```